apps/prairielearn/src/tests/sprocs/check_assessment_access.test.ts

```diff
@@ -1,59 +1,225 @@
+      const authorized = await checkAssessmentAccess({
+        assessment_id: '50',
+        authz_mode: 'Exam',
+        course_role: 'None',
+        course_instance_role: 'None',
+        user_id: '1000',
+        uid: 'valid@school.edu',
+        date: '2012-07-07 06:06:06-00',
+        display_timezone: 'US/Central',
       });
-      describe('Linked exam', () => {
-        reservationTests(11, 1, 13, false);
+      assert.isFalse(authorized);
+    });
+
+    it('should not allow access when mode:Public and exam_uuid is present', async () => {
+      const authorized = await checkAssessmentAccess({
+        assessment_id: '52',
+        authz_mode: 'Public',
+        course_role: 'None',
+        course_instance_role: 'None',
+        user_id: '1000',
+        uid: 'valid@school.edu',
+        date: '2010-07-07 06:06:06-00',
+        display_timezone: 'US/Central',
       });
-      describe('Linked exam in different PS course', () => {
-        reservationTests(12, 5, 13, false);
+      assert.isFalse(authorized);
+    });
+  });
+
+  /////////////////////////////////////////////////////////////////////////////
+
+  describe('check_assessment_access PrairieTest scheduler tests', function () {
+    describe('No checked in reservation', () => {
+      it('should allow access to an exam without exam_uuid', async () => {
+        const authorized = await checkAssessmentAccess({
+          assessment_id: '10',
+          authz_mode: 'Exam',
+          course_role: 'None',
+          course_instance_role: 'None',
+          user_id: '1000',
+          uid: 'valid@school.edu',
+          date: '2010-07-07 06:06:06-00',
+          display_timezone: 'US/Central',
+        });
+        assert.isTrue(authorized);
+      });
+
+      it('should not allow access to an exam with exam_uuid', async () => {
+        const authorized = await checkAssessmentAccess({
+          assessment_id: '11',
+          authz_mode: 'Exam',
+          course_role: 'None',
+          course_instance_role: 'None',
+          user_id: '1000',
+          uid: 'valid@school.edu',
+          date: '2010-07-07 06:06:06-00',
+          display_timezone: 'US/Central',
+        });
+        assert.isFalse(authorized);
+      });
+    });
+
+    describe('Checked in reservation', () => {
+      before(`create checked-in reservation for student`, async () => {
+        await sqldb.queryAsync(sql.insert_pt_reservation, { exam_id: 1 });
+      });
+
+      // it('should not allow access to an exam without exam_uuid', async () => {
+      //   const authorized = await checkAssessmentAccess({
+      //     assessment_id: '10',
+      //     authz_mode: 'Exam',
+      //     course_role: 'None',
+      //     course_instance_role: 'None',
+      //     user_id: '1000',
+      //     uid: 'valid@school.edu',
+      //     date: '2010-07-07 06:06:06-00',
+      //     display_timezone: 'US/Central',
+      //   });
+      //   assert.isFalse(authorized);
+      // });
```

Comment:
@trombonekenny from what I can tell, this test tries to assert behavior that doesn't currently exist. Let's discuss. I don't think we necessarily want the behavior that this test expects, as for better or for worse we still support legacy `exam_mode_networks` independently of PT and `exam_uuid`.

Reply:
Is there an edge case where someone is in mode: Exam and there's an exam with mode: Exam protections but no exam_uuid?

Reply:
I wouldn't call it an edge case, but yes, that's what would happen if someone is using `exam_mode_networks`, which some self-hosters are.

Reply:
Or are you thinking about the case where a student is in a PT center and they end up with access to one exam with an `examUuid` and one without?

Reply:
Yeah, I was looking for tests for all the possible permutations of access like mode: Exam, checked in reservation, exam protected by uuid (and their not counterparts.)

Reply:
I agree that it's useful to have assertions in place; the question is what behavior we actually want to assert. Exam-mode assessments without UUIDs is indeed a thing we support, so I think we should assert that access is allowed in that case.

Reply:
At the dev meeting today we concluded that we do care about the case when we are in exam mode but don't have an examUuid. However, to handle this case correctly we'll need to remember whether we are in exam mode because of PT or because of exam mode networks. Passing this information down to the assessment access checks will require passing down an extra variable.